challenges/sre/challenge-security.md

```diff
@@ -0,0 +1,202 @@
+
+### Service Levels & Deployment
+
+After the server code is approved, submit a final PR including service levels, deployment, and any remaining requirements.
+
+Your chosen service levels should focus on meaningful indicators, be set at realistic thresholds, and align with the system's functionality and user experience.  The presentation of these SLOs is up to the candidate's discretion.  Consider a simple solution and include comments on how a more robust solution could be implemented (see [trade-offs](#trade-offs)).
+
+When selecting [security invariants](https://d1.awsstatic.com/events/aws-reinforce-2022/NIS234_Learn-to-create-end-to-end-security-invariants-using-AWS-services.pdf), consider which properties you expect your system to always maintain.  The chosen invariants should cover attack vectors against the most critical parts of your system design. How these properties are evaluated and reported is up to the candidate.
+
+This part of the implementation will be evaluated ensuring the helm chart includes production-level packaging, service to service communication is secured implementing zero-trust, all artifacts are easy to build and deploy, and that the selected security invariants of the system can be evaluated.
+
+### Tooling
+For evaluation purposes, your solution should be written in Go and is deployable to a local Kubernetes cluster. The choice of which local Kubernetes cluster is up to you, but please ensure compatibility with both macOS and Linux. We suggest [KIND](https://kind.sigs.k8s.io/).
+
+You may use additional external dependencies, but ensure that detecting or installing these are straightforward for the reviewer. At a minimum, your solution should include the following dependencies:
+
+* Go
+* make
+* Docker
+
+## Level 4
+
+### Server
+* HTTPS API to create, get, or list users with Redis datastore
+* HTTPS API to add a new payment with Redis datastore
+* HTTPS API to get payments by user with Redis datastore
+* HTTPS Health check ensuring availability
+* Create a secure client API with mTLS
+* Include one or two tests that cover happy and unhappy path scenarios
+
+### Service Levels
+* Define three Service Level Objectives (SLOs) for the API service
+* Create a service to periodically evaluate the defined SLOs and report on their status
+* SLO service communication is secure
+
+### Deployment
+* Create a Helm chart for the API server
+* Include the ability to build and deploy all artifacts and certificates to a Kubernetes cluster using make
+* Ensure artifacts and services are production ready
+* Evaluate the state of three security invariants.
+
+## Level 5
```

Comment:
We're going to need more than gRPC, mTLS, autoscaling, and load testing. In some ways gRPC makes it easier. And I'd rather have mTLS for L4 (maybe even L3). I want to test zero-trust knowledge. That's our game.

Maybe we increase the complexity by having them deploy a Prom/Loki/Promtail stack, increase the SLOs & security invariants to 5, and say it all needs to be zero-trust. I'd rather make it so hard people won't even try than too easy and we waste everyone's time. Much easier on everyone to reduce the requirements if they turn out to be too much.

At least with the prom/loki stack, some interesting invariant and SLO tracking is possible. We may even consider handing them a bit of the config. But in the end, I would love to hire an L5 who could set all this up in a few weeks. That would mean they will be able to contribute right away.

Reply:
We could even make silly things more obvious by saying a restart of the kind server doesn't preserve data.

Reply:
mTLS should be required in both levels fwiw.  

I'll think on how to make the L5 bar a bit higher.